File: problems/680/solution.go
package main
func isPalindrome(s string, i, j int) bool {
    for i < j {
        if s[i] != s[j] {
            return false
        }
        i++
        j--
    }
    
    return true
}

func validPalindrome(s string) bool {
    for i, j := 0, len(s) - 1; i < j; {
        if (s[i] != s[j]) {
            return isPalindrome(s, i + 1, j) || isPalindrome(s, i, j - 1)
        }
        
        i++
        j--
    }
    
    return true
}

File: problems/680/solution.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        int lo = 0, hi = size(s) - 1;
        
        while (lo <= hi) {
            if (s[lo] != s[hi]) {
                return isPalindrome(s, lo + 1, hi) or isPalindrome(s, lo, hi - 1);
            }
            
            lo++;
            hi--;
        }
        
        return true;
    }
    
    bool isPalindrome(const string& s, int lo, int hi) {
        while (lo <= hi) {
            if (s[lo++] != s[hi--]) {
                return false;
            }
        }
        
        return true;
    }
};